server/dump.go

```diff
@@ -19,8 +19,14 @@ func (p *Plugin) createDump(ctx context.Context, min, max time.Time, remoteStora
 		return "", errors.New("no blocks in the remote sotrage")
 	}
 
-	zipFileNameRemote := filepath.Join(pluginDataDir, PluginName, zipFileName)
-	dumpDir := filepath.Join(PluginName, "dump")
+	// we generate everything under a new directory to avoid conflicts
+	// between simultaneous downloads
+	tempDir := model.NewId()
+	tempZipFile := tempDir + zipFileName
```

Comment:
the tempDir should be enough. Is there a need to add this to the filename as well? Duplicate files downloaded on the browser automatically get a ` (1)` appended to the end.

Reply:
You're right. I did some improvements as well. See: https://github.com/mattermost/mattermost-plugin-metrics/pull/18/commits/fbd55119c820262e087d03c0342391d6ac6c4014